packages/api/lambdas/process-s3-dead-letter-archive.js

```diff
@@ -1,58 +1,58 @@
 'use strict';
 
-const moment = require('moment');
-
+//@ts-check
 const log = require('@cumulus/common/log');
 
 const S3 = require('@cumulus/aws-client/S3');
 const { s3 } = require('@cumulus/aws-client/services');
 const { getJsonS3Object, deleteS3Object } = require('@cumulus/aws-client/S3');
 const { getKnexClient } = require('@cumulus/db');
-const { unwrapDeadLetterCumulusMessage } = require('@cumulus/message/DeadLetterMessage');
+const {
+  unwrapDeadLetterCumulusMessage,
+  getDLAKey,
+} = require('@cumulus/message/DeadLetterMessage');
 
 const { writeRecords } = require('./sf-event-sqs-to-db-records');
-
+/**
+ *
+ * @typedef {import('@cumulus/types/api/dead_letters').DLARecord} DLARecord
+ */
 /**
  * Generates new archive key for unprocessed dead letter message
  *
- * @param {string}   [failedKey] - key of message that failed to process
+ * @param {DLARecord} failedMessage
  * @returns {string}
  */
-const generateNewArchiveKeyForFailedMessage = (failedKey) => {
-  // Split key with format `${process.env.stackName}/dead-letter-archive/sqs/${messageId}`,
-  const splitKey = failedKey.split('/sqs/');
-  const pathPrefix = splitKey[0];
-  const messageId = splitKey[1];
-  const date = moment.utc().format('YYYY-MM-DD');
-  return `${pathPrefix}/failed-sqs/${date}/${messageId}`;
-};
+const generateNewArchiveKeyForFailedMessage = (failedMessage) => (
+  getDLAKey(process.env.stackName, failedMessage).replace('sqs/', 'failed-sqs/')
```

Comment:
Can your replace `/dead-letter-archive/sqs/` to be safe?

Reply:
done